### PR TITLE
Add SimpleCov

### PR DESCRIPTION
**Pull request by andyw8**
### Motivation

I was trying out SimpleCov a while ago, but I figured we may as well merge the branch since it can sometimes by handy to look at the current coverage. I made it opt-in since the overhead from running tests with coverage enabled is:

**Main:** bundle exec rake  50.53s user 9.65s system 82% cpu 1:12.59 total
**This branch:** bundle exec rake  69.23s user 8.92s system 86% cpu 1:30.70 total

(It writes to `coverage/` which is already in `.gitignore`).

### CODE REVIEW

File: test/fixtures/prism
Comment (vinistock) [inline test/fixtures/prism:1]: Can we remove these fixture updates?

File: test/test_helper.rb [hunk 1]
```diff
@@ -1,6 +1,16 @@
 # typed: strict
 # frozen_string_literal: true
 
+require "sorbet-runtime"
```
[the remark this answers is not in the document]
Comment (vinistock) [reply test/test_helper.rb:4]: We get it when we require `ruby_lsp/internal`. Can we move the `SimpleCov.start` below that require?

Or is the recommendation that this needs to be required before anything else?

File: test/test_helper.rb [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+
+if ENV["COVERAGE"]
+  require "simplecov"
+
+  SimpleCov.start do
+    T.bind(self, SimpleCov::Configuration)
+    enable_coverage :branch
```
Comment (vinistock) [inline test/test_helper.rb:11]: Out of curiosity, what does this do?

File: test/test_helper.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (st0012) [reply test/test_helper.rb:4]: IIRC SimpleCov needs to be required and activated before the target (in this case, `ruby_lsp/internal`) is required. So yeah, we probably need to do this.

File: test/test_helper.rb (hunk 2 above)
Comment (st0012) [reply test/test_helper.rb:11]: It checks condition coverage too (versus just line coverage), which is particularly useful for inline conditions like `foo if bar?`

Reference: https://github.com/simplecov-ruby/simplecov?tab=readme-ov-file#branch-coverage-ruby--25

**Review by st0012 — COMMENTED** [review]
Are we able to use any coverage Saas (e.g. https://about.codecov.io/)? Otherwise, I feel we may underutilise this tool because it takes several steps to see it locally.

**Review by st0012 — APPROVED** [review]

**Pull request merged**